nano/lib/config.hpp

```diff
@@ -138,35 +138,55 @@ class work_thresholds
 class network_constants
 {
 public:
-	network_constants (nano::work_thresholds & work, nano::networks network_a) :
-		current_network (network_a),
-		work{ work }
+	network_constants (nano::work_thresholds & work_, nano::networks network_a) :
+		current_network(network_a),
+		work(work_),
+		principal_weight_factor(1000), // 0.1% A representative is classified as principal based on its weight and this factor
+		default_node_port(44000),
+		default_rpc_port(45000),
+		default_ipc_port(46000),
+		default_websocket_port(47000),
+		request_interval_ms(500),
+		cleanup_period(std::chrono::seconds(60)),
+		idle_timeout(std::chrono::seconds(120)),
```

Comment:
`idle_timeout` is slightly incorrect, it should be:
```
cleanup_period * 2
```

Reply:
I understand the original code is cleanup_period * 2
the default value I did do std::chrono::seconds(120) which is cleanup_period(std::chrono::seconds(60)) * 2
so the value is correct. 

The reason I did it like this is that this is in the initialization list, the order is not initialized based on there order, but based on the order that they in the header and with out a good understanding of c++ that is something easy to miss. 

Some one could refactor the member variables and break things with no clear understanding why cause there is no direct linkage.

do you think having something outside the class, like default cleanup period, then have
		cleanup_period(default_cleanup_period)),
		idle_timeout(default_cleanup_period * 2),

Reply:
Let's do the initialising inside the constructor. I agree that the initialiser list could be tricky and a potential source of bugs.